### PR TITLE
genie: add build_type + allow cross build

### PR DESCRIPTION
**Pull request by madebr**
Hey @intelliglide

This patch allows me to crossbuild genie, and build genie in debug and release mode.

### CODE REVIEW

File: recipes/genie/all/conanfile.py [hunk 1]
```diff
@@ -66,17 +74,15 @@ def build(self):
...
+            autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
             with tools.chdir(self._source_subfolder):
-                self.run(make, win_bash=tools.os_info.is_windows)
+                autotools.make(args=["OS={}".format(self._os), "config={}".format(self._genie_config)])
 
     def package(self):
         self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
         bin_ext = ".exe" if self.settings.os == "Windows" else ""
         self.copy("genie{}".format(bin_ext), dst="bin", src=os.path.join(self._source_subfolder, "bin", self._os))
+        self.copy("*.lua", dst="res", src=os.path.join(self._source_subfolder, "src"))
```
Comment (intelligide) [inline recipes/genie/all/conanfile.py]: genie doesn't need lua scripts on non-debug configs.

Comment (madebr) [inline recipes/genie/all/conanfile.py]: I copied them just to be safe.
```suggestion
        if self.settings.build_type == "Debug":
            self.copy("*.lua", dst="res", src=os.path.join(self._source_subfolder, "src"))
```

File: recipes/genie/all/conanfile.py [hunk 2]
```diff
@@ -85,3 +91,7 @@ def package_info(self):
         bindir = os.path.join(self.package_folder, "bin")
         self.output.info("Appending PATH environment variable: {}".format(bindir))
         self.env_info.PATH.append(bindir)
+
+        resdir = os.path.join(self.package_folder, "res")
+        self.output.info("Appending PREMAKE_PATH environment variable: {}".format(resdir))
+        self.env_info.PREMAKE_PATH.append(resdir)
```
Comment (madebr) [inline recipes/genie/all/conanfile.py]:
```suggestion
        if self.settings.build_type == "Debug":
            resdir = os.path.join(self.package_folder, "res")
            self.output.info("Appending PREMAKE_PATH environment variable: {}".format(resdir))
        self.env_info.PREMAKE_PATH.append(resdir)
```

File: recipes/genie/all/conanfile.py [hunk 3]
```diff
@@ -85,3 +92,8 @@ def package_info(self):
         bindir = os.path.join(self.package_folder, "bin")
         self.output.info("Appending PATH environment variable: {}".format(bindir))
         self.env_info.PATH.append(bindir)
+
+        if self.settings.build_type == "Debug":
+            resdir = os.path.join(self.package_folder, "res")
+            self.output.info("Appending PREMAKE_PATH environment variable: {}".format(resdir))
+        self.env_info.PREMAKE_PATH.append(resdir)
```
Comment (madebr) [inline recipes/genie/all/conanfile.py]:
```suggestion
            self.output.info("Appending PREMAKE_PATH environment variable: {}".format(resdir))
            self.env_info.PREMAKE_PATH.append(resdir)
```

**Pull request merged**